react/components/utils/withDeviceHoc.tsx

```diff
@@ -8,18 +8,23 @@ export enum Device {
 }
 
 const withDevice = Component => {
-  function WithDevice(props) {
+  function WithDevice({
+    device: defaultIsDevice,
+    isMobile: defaultIsMobile,
+    ...props
+  }) {
     // Taken at https://github.com/vtex-apps/device-detector/blob/master/react/useDevice.tsx#L22
     const isScreenMedium = useMedia({ minWidth: '40rem' })
     const isScreenLarge = useMedia({ minWidth: '64.1rem' })
 
-    const device = isScreenLarge
-      ? Device.desktop
-      : isScreenMedium
-      ? Device.tablet
-      : Device.phone
+    const device =
+      defaultIsDevice ?? isScreenLarge
```

Comment:
If it's supposed to be a default value, shouldn't it be on the right side of the operation?
Also, why is a string being evaluated as a boolean value?

Reply:
I think that i used the wrong word to name this prop, is more to `customDevice` and `customIsMobile`, since it is passed through the `FilterBar` component

Reply:
It isn't evaluated as a boolean, it checks if there is a defined `customDevice`, otherwise, it will return some type of the `Device` enum

Reply:
Now I get it. Much better with the `custom*` thing.